dora/src/bytecode/generator.rs

```diff
@@ -488,6 +488,53 @@ impl<'a, 'ast> AstBytecodeGen<'a, 'ast> {
         }
     }
 
+    fn visit_expr_delegation(&mut self, expr: &ExprDelegationType, dest: DataDest) -> Register {
+        let call_type = self.src.map_calls.get(expr.id).unwrap().clone();
+        let fct_id = call_type.fct_id().unwrap();
+
+        let callee_id = fct_id;
+        let callee = self.vm.fcts.idx(callee_id);
+        let callee = callee.read();
+
+        let return_type = if dest.is_effect() {
```

Comment:
Constructors do not return a value, so we can simply assert `callee.return_type.is_unit()` here instead

Reply:
I already added the assert, but I missed to refactor this. Should be fixed.

Reply:
I added also an assertion for `dest.is_effect()` to be pedantic. cf. `visit_expr_assign`.